### PR TITLE
bpo-28230: Document the pathlib support in tarfile and add tests.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Lib/test/test_tarfile.py [hunk 1]
```diff
@@ -440,6 +441,22 @@ def test_bytes_name_attribute(self):
                 self.assertIsInstance(tar.name, bytes)
                 self.assertEqual(tar.name, os.path.abspath(fobj.name))
 
+    def test_pathlike_name(self):
+        tarname = pathlib.Path(self.tarname)
+        with tarfile.open(tarname, mode=self.mode) as tar:
+            self.assertIsInstance(tar.name, str)
+            self.assertEqual(tar.name, os.path.abspath(str(tarname)))
```
Comment (berkerpeksag) [inline Lib/test/test_tarfile.py]: Perhaps use `os.fspath` instead of `str`?

**Review by berkerpeksag — APPROVED** [review]
It looks pretty good to me, thanks! All documented members are correctly tested and tests are passed in my local environment too.

You may want to document this in `Misc/NEWS`. Side note: We probably should tweak the 3.6 whatsnew document to give readers up-to-date information about pathlib support in the standard library.

I just left a super minor comment.

File: Lib/test/test_tarfile.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_tarfile.py]: Done.

**Comment by serhiy-storchaka** [comment]
I have not added an entry in `Misc/NEWS` because this change doesn't change the behavior. It just documents existing feature. I'm not sure about What's New document. Path-like support in `tarfile` (as well as in many other stdlib and third-party modules) is just a side effect of implementing PEP 519 in `os` and `os.path`.

**Comment by Mariatta** [comment]
I removed the `needs backport to 3.6 label` since it was done.

**Pull request merged**